front/components/DataSourceViewPermissionTree.tsx

```diff
@@ -0,0 +1,66 @@
+import type {
+  ConnectorPermission,
+  ContentNodesViewType,
+  DataSourceViewType,
+  LightWorkspaceType,
+} from "@dust-tt/types";
+
+import type { ContentNodeTreeNodeStatus } from "@app/components/ContentNodeTree";
+import { ContentNodeTree } from "@app/components/ContentNodeTree";
+import { useDataSourceViewContentNodes } from "@app/lib/swr/data_source_views";
+
+interface DataSourceViewPermissionTreeProps {
+  dataSourceView: DataSourceViewType;
+  displayDocumentSource: (documentId: string) => void;
+  isSearchEnabled?: boolean;
+  isRoundedBackground?: boolean;
+  owner: LightWorkspaceType;
+  parentId?: string | null;
+  permissionFilter?: ConnectorPermission;
+  showExpand?: boolean;
+  viewType: ContentNodesViewType;
+  treeSelectionModel?: Record<string, ContentNodeTreeNodeStatus>;
+  setTreeSelectionModel?: (
+    updater: (
+      prev: Record<string, ContentNodeTreeNodeStatus>
+    ) => Record<string, ContentNodeTreeNodeStatus>
+  ) => void;
+}
+
+export function DataSourceViewPermissionTree({
+  dataSourceView,
+  isSearchEnabled,
+  isRoundedBackground,
+  owner,
+  displayDocumentSource,
+  showExpand,
+  viewType,
+  treeSelectionModel,
+  setTreeSelectionModel,
+}: DataSourceViewPermissionTreeProps) {
+  const useResourcesHook = (parentId: string | null) => {
```

Comment:
Actually using useCallback (even if it would be ok) breaks the rules-of-hook (React Hook "usePokeConnectorPermissions" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.). 
Defining a dynamic hook a in render function is not a violation of any hook rule so should not be blocked by a linter ?